Implement bitCount intrinsic using SIMD instructions for AArch64

armLeiJin opened the pull request:
1. Implement 3 new Neon instructions for AArch64, including cnt, addv, umov
2. Use fmov, cnt, addv, umov to implement bitcount intrinsic for Aarch64
3. Fix nodeinfo for bitcount node in Aarch64 since it is different from
other platforms
4. Add instructions encoding test for newly added instructions and code
gen test for bitcount node instrinsic modification

Change-Id:I8437d12e8c9dfdf354a8480d4f4a697f75eb9ab9

armLeiJin commented:
@sanzinger,  could you take a review on the patch? Thanks!

sanzinger commented on compiler/src/org.graalvm.compiler.replacements.aarch64/src/org/graalvm/compiler/replacements/aarch64/AArch64GraphBuilderPlugins.java:
```diff
@@ -64,8 +64,8 @@ public static void register(Plugins plugins, BytecodeProvider bytecodeProvider,
         invocationPlugins.defer(new Runnable() {
             @Override
             public void run() {
-                registerIntegerLongPlugins(invocationPlugins, AArch64IntegerSubstitutions.class, JavaKind.Int, bytecodeProvider);
```
This would make the `AArch64IntegerSubstitutions` and `AArch64LongSubstitutions` obsolete right?

sanzinger commented on compiler/src/org.graalvm.compiler.replacements.aarch64/src/org/graalvm/compiler/replacements/aarch64/AArch64GraphBuilderPlugins.java:
```diff
@@ -64,8 +64,8 @@ public static void register(Plugins plugins, BytecodeProvider bytecodeProvider,
             public void run() {
-                registerIntegerLongPlugins(invocationPlugins, AArch64IntegerSubstitutions.class, JavaKind.Int, bytecodeProvider);
-                registerIntegerLongPlugins(invocationPlugins, AArch64LongSubstitutions.class, JavaKind.Long, bytecodeProvider);
+                registerIntegerLongPlugins(invocationPlugins, JavaKind.Int, bytecodeProvider);
```
This would make the `AArch64IntegerSubstitutions` and `AArch64LongSubstitutions` obsolete right? So we could delete both classes?

armLeiJin replied:
Yes, you are right.

sanzinger commented:
FYI, I have rebased the PR to latest `master` and removed `AArch64IntegerSubstitutions` and AArch64LongSubstitutions`.

armLeiJin commented:
> FYI, I have rebased the PR to latest `master` and removed `AArch64IntegerSubstitutions` and AArch64LongSubstitutions`. Please give me an OK if this correct.

OK. Thanks!
